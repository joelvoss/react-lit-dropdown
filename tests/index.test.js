import * as React from 'react';
import { render, userEvent } from './test-utils';
import { axe } from 'jest-axe';

import {
	DropdownProvider,
	DropdownTrigger,
	DropdownPopover,
	DropdownItems,
	DropdownItem,
} from '../src/index';
import { act } from 'react-dom/test-utils';

describe('Dropdown', () => {
	const Comp = () => {
		const [selectedItem, selectedItemSet] = React.useState('');
		const handleSelect = item => {
			selectedItemSet(item);
		};

		return (
			<div>
				<DropdownProvider>
					<DropdownTrigger id="actions-button">
						Actions <span aria-hidden="true">â–¾</span>
					</DropdownTrigger>
					<DropdownPopover>
						<DropdownItems>
							<DropdownItem onSelect={() => handleSelect('item 1')}>
								Dropdown Item
							</DropdownItem>
							<hr />
							<DropdownItem isLink as="a" href="#">
								Dropdown Item Link
							</DropdownItem>
						</DropdownItems>
					</DropdownPopover>
				</DropdownProvider>
				<p data-testid="click-result">Selected item: {selectedItem}</p>
			</div>
		);
	};

	it('should not have ARIA violations', async () => {
		let { container, queryByText } = render(<Comp />);
		await act(async () => {
			expect(await axe(container)).toHaveNoViolations();
		});
		userEvent.click(queryByText(/Actions/i));
		await act(async () => {
			expect(await axe(container)).toHaveNoViolations();
		});
	});

	it('should render proper HTML', async () => {
		const { baseElement, queryByText } = render(<Comp />);
		expect(baseElement).toMatchSnapshot();
		userEvent.click(queryByText(/Actions/i));
		expect(baseElement).toMatchSnapshot();
	});
});
