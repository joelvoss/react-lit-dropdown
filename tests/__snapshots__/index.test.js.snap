// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should render proper HTML 1`] = `
<body>
  <div>
    <div>
      <button
        id="actions-button"
        type="button"
      >
        Actions 
        <span
          aria-hidden="true"
        >
          ▾
        </span>
      </button>
      <p
        data-testid="click-result"
      >
        Selected item: 
      </p>
    </div>
  </div>
  <portal>
    <div
      hidden=""
      style="position: absolute; display: none; left: 0px; top: 0px;"
    >
      <div
        id="menu--7"
        style="display: block; white-space: nowrap; border: 1px solid hsla(0, 0%, 0%, 0.25); background: rgb(255, 255, 255);"
        tabindex="-1"
      >
        <div
          data-valuetext="Dropdown Item"
          id="option-0--menu--7"
          style="display: block; user-select: none; cursor: pointer; padding: 5px 20px;"
          tabindex="-1"
        >
          Dropdown Item
        </div>
        <hr />
        <a
          data-valuetext="Dropdown Item Link"
          href="#"
          id="option-1--menu--7"
          style="display: block; user-select: none; cursor: pointer; padding: 5px 20px;"
          tabindex="-1"
        >
          Dropdown Item Link
        </a>
      </div>
    </div>
  </portal>
</body>
`;

exports[`Dropdown should render proper HTML 2`] = `
<body>
  <div>
    <div>
      <button
        id="actions-button"
        type="button"
      >
        Actions 
        <span
          aria-hidden="true"
        >
          ▾
        </span>
      </button>
      <p
        data-testid="click-result"
      >
        Selected item: 
      </p>
    </div>
  </div>
  <portal>
    <div
      style="position: absolute; display: block; left: 0px; top: 0px;"
    >
      <div
        id="menu--7"
        style="display: block; white-space: nowrap; border: 1px solid hsla(0, 0%, 0%, 0.25); background: rgb(255, 255, 255);"
        tabindex="-1"
      >
        <div
          data-valuetext="Dropdown Item"
          id="option-0--menu--7"
          style="display: block; user-select: none; cursor: pointer; padding: 5px 20px;"
          tabindex="-1"
        >
          Dropdown Item
        </div>
        <hr />
        <a
          data-valuetext="Dropdown Item Link"
          href="#"
          id="option-1--menu--7"
          style="display: block; user-select: none; cursor: pointer; padding: 5px 20px;"
          tabindex="-1"
        >
          Dropdown Item Link
        </a>
      </div>
    </div>
  </portal>
</body>
`;
